src/yaml.cpp

```diff
@@ -104,30 +103,132 @@ estd::result<YAML::Node, estd::error> readYamlFile(std::string const & path) {
+YamlResult<void> mergeYamlMaps(YAML::Node & map_a, YAML::Node map_b) {
+	if (map_b.IsMap()){
+		for (YAML::const_iterator iterator = map_b.begin(); iterator != map_b.end(); iterator++) {
+			std::string key = iterator->first.as<std::string>();
+			YAML::Node value = iterator->second;
+			if (map_a[key] && map_a[key].IsMap()) {
+				auto merged = mergeYamlMaps(map_a[key], value);
+				if (!merged) {
+					merged.error().appendTrace({key, "", YAML::NodeType::Map});
+					return merged;
+				}
+			}
+			else if (value.Tag() == "!ordered_dict"  && map_a[key].Tag() == "!ordered_dict" && map_a[key]) {
+				auto merged = mergeYamlOrderedDict(map_a[key], value);
+				if (!merged) {
+					merged.error().appendTrace({key, "", YAML::NodeType::Map});
+					return merged;
+				}
+			}
+			else {
+			map_a[key] = value;
+			}
+		}
+	}
+	return estd::in_place_valid;
+}
 
-YamlResult<void> mergeYamlNodes(YAML::Node & map_a, YAML::Node map_b) {
-	// Check if the arguments are maps.
-	if (!map_a.IsMap() && !map_a.IsNull()) {
-		return YamlError{"tried to merge into a YAML node that is not a map"};
+// Uitility function to check whether a value is present in a vector.
+bool Contains(const std::vector<int> &list, int x){
+	return std::find(list.begin(), list.end(), x) != list.end();
+}
+
+// Check whether item is a single item map.
+bool singleMap(YAML::Node yaml_node){
+	if (!yaml_node.IsMap() && yaml_node.size() > 1){
+		return false;
 	}
-	if (!map_b.IsMap() && !map_b.IsNull()) {
-		return YamlError{"tried to merge from a YAML node that is not a map"};
+	return true;
+}
+// Utility function to check whether key in map_b is present in map_a.
+std::optional<std::size_t> orderedDictFind(YAML::Node map_a, std::string const & key){
+	for (std::size_t j = 0; j < map_a.size(); j++){
+			if (map_a[j][key]){
+					return j;
+				}
+			
 	}
+	
+	return std::nullopt;
+}
 
-	for (YAML::const_iterator iterator = map_b.begin(); iterator != map_b.end(); iterator++) {
-		std::string key = iterator->first.as<std::string>();
-		YAML::Node value = iterator->second;
-		if (value.IsMap() && map_a[key]) {
-			auto merged = mergeYamlNodes(map_a[key], value);
-			if (!merged) {
-				merged.error().appendTrace({key, "", YAML::NodeType::Map});
-				return merged;
+// Merge ordered dictionaries.
+YamlResult<void> mergeYamlOrderedDict(YAML::Node & map_a, YAML::Node map_b){
+	// Vector containing all the indexes of b which are present in a.
+	std::vector<int> checked;
+	std::optional<std::size_t> index;
+	bool single_map = true;
+	// Boolean to determine whether the element in b is present in a or not.
+	bool found = false;
+	for (std::size_t i = 0; i < map_b.size(); i++) {
+		found = false;
+		// Check whether the element in b is a single key-value pair.
+		if (!singleMap(map_b[i])){
+			return YamlError("Ordered dictionary should only contain single item map");
+		}
+		for (YAML::const_iterator iterator = map_b[i].begin(); iterator != map_b[i].end(); iterator++){
```

Comment:
This shouldn't be a loop now, right? You've already checked that `map_b[i]` is a single-item map, so there is only one child.